app/models/Property.test.ts

```diff
@@ -1,56 +1,183 @@
+
+    it("correctly calculates newComponentValue for foundations", () => {
+      const result = property.calculateComponentValue(
+        'foundations',
+        property.newBuildPrice,
+        property.age,
+        MAINTENANCE_LEVELS[0]
+      );
+
+      const expectedNewComponentValue = 
+        property.newBuildPrice * 
+        HOUSE_BREAKDOWN_PERCENTAGES.foundations.percentageOfHouse;
+
+      expect(result.newComponentValue).toBe(expectedNewComponentValue);
+      expect(result.maintenanceAddition).toBe(0); // Foundations should have no maintenance
+    });
+
+    it("correctly calculates depreciationFactor for internal linings", () => {
+      const result = property.calculateComponentValue(
+        'internalLinings',
+        property.newBuildPrice,
+        property.age,
+        MAINTENANCE_LEVELS[0]
+      );
+
+      const expectedDepreciationFactor = 
+        1 - (HOUSE_BREAKDOWN_PERCENTAGES.internalLinings.depreciationPercentageYearly * property.age);
+
+      expect(result.depreciationFactor).toBe(expectedDepreciationFactor);
+    });
+
+    it("correctly calculates maintenanceAddition for electrical appliances", () => {
+      const result = property.calculateComponentValue(
+        'electricalAppliances',
+        property.newBuildPrice,
+        property.age,
+        MAINTENANCE_LEVELS[0]
+      );
+
+      const expectedMaintenanceAddition = 
+        MAINTENANCE_LEVELS[0] * 
+        property.newBuildPrice * 
+        property.age * 
+        HOUSE_BREAKDOWN_PERCENTAGES.electricalAppliances.percentOfMaintenanceYearly;
+
+      expect(result.maintenanceAddition).toBe(expectedMaintenanceAddition);
+    });
+
+    it("correctly calculates depreciatedComponentValue for ventilation services", () => {
+      const result = property.calculateComponentValue(
+        'ventilationServices',
+        property.newBuildPrice,
+        property.age,
+        MAINTENANCE_LEVELS[0]
+      );
+
+      const component = HOUSE_BREAKDOWN_PERCENTAGES.ventilationServices;
+      const newComponentValue = property.newBuildPrice * component.percentageOfHouse;
+      const depreciationFactor = 1 - (component.depreciationPercentageYearly * property.age);
+      const maintenanceAddition = 
+        MAINTENANCE_LEVELS[0] * 
+        property.newBuildPrice * 
+        property.age * 
+        component.percentOfMaintenanceYearly;
+
+      const expectedValue = Math.max(
+        (newComponentValue * depreciationFactor) + maintenanceAddition,
+        0
+      );
+
+      expect(result.depreciatedComponentValue).toBe(expectedValue);
+    });
+
+    it("ensures depreciatedComponentValue never goes below 0", () => {
+      const result = property.calculateComponentValue(
+        'ventilationServices',
+        property.newBuildPrice,
+        100, // High age to test possible negative values
+        MAINTENANCE_LEVELS[0]
+      );
+
+      expect(result.depreciatedComponentValue).toBeGreaterThanOrEqual(0);
+    });
```

Comment:
In tests like this we can use something like `test.each()` ([docs](https://jestjs.io/docs/api#testeachtablename-fn-timeout))

It might be more meaningful to have the same test, with a number of variable ages, to prove we have a floor at 0.